lib/services/peer_info_service.dart

```diff
@@ -20,23 +20,62 @@ class PeerInfoService with ChangeCallbackProvider {
           .where((peerInfo) => peerInfo.id != null && peerInfo.name.isNotEmpty)
           .map((peerInfo) => MapEntry(peerInfo.id!, peerInfo.name)));
 
-  Future<void> upsert(PeerInfo peerInfo) async {
-    await _repository.upsert(peerInfo);
+  Future<void> upsert(
+    PeerInfo peerInfo, {
+    bool mergeWithExistent = true,
```

Comment:
this should always be true? mergeWithExistent can be removed and just checking for peerInfo.id != null :)